### PR TITLE
chore: add devcontainer for better dev experience

### PR DESCRIPTION
**Pull request by tisonkun**
### What changes were proposed in this pull request?

Add configs for [devcontainer](https://devcontainers.github.io/implementors/json_reference/) so that developers can leverage [VS Code DevContainers](https://code.visualstudio.com/docs/devcontainers/containers), [GitHub CodeSpace](https://github.com/codespaces), and so on.

### Why are the changes needed?

For developing OceanBase on any machine and even in browser.

### Will break the compatibility? How if so?

NO

### Does this PR introduce any user-facing change?

NO

### How was this patch tested?

~~I'll attach a video later for how this feature works.~~

You can view at https://www.bilibili.com/video/BV1pL411Y74a.

### Checklist

- [x] I've run the tests to see all new and existing tests pass.
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above.
- [x] I've informed the technical writer about the documentation change if necessary.

### CODE REVIEW

File: deps/init/dep_create.sh [hunk 1]
```diff
@@ -100,6 +100,9 @@ function get_os_release() {
         version_ge "8.0" && OS_RELEASE=8 && return
         version_ge "7.0" && OS_RELEASE=7 && return
         ;;
+      ubuntu)
+        version_ge "16.04" && compat_centos7 && return
```
Comment (tisonkun) [inline deps/init/dep_create.sh:104]: Verified locally:

image

image

**Comment by caifeizhi** [comment]
Thank you for your contribution!
I just tried devcontainer and it can run and compile normally, so this PR is okay. 
However, I encountered two issues during the use process of a container with 4C8G (mem) and 32G (disk), which need to be resolved later: 
1. The compilation process still takes a relatively long time. Originally, it was expected to compile in 15 minutes on a machine with 4C8G, but the actual runtime exceeded half an hour.
2. The deployment check requires more than 54G of free disk space, but after compiling, there is only about 12G of disk space left, which is not enough for correct deployment and operation. 

Therefore, it may be necessary for the user to open a higher spec codespace(maybe 8C16G + 64G) at present.

**Comment by tisonkun** [comment]
> The compilation process still takes a relatively long time. Originally, it was expected to compile in 15 minutes on a machine with 4C8G, but the actual runtime exceeded half an hour.

Cloud resources can be oversold. Locally it takes about 20 minutes to compile. I don't think it's a significant blocker as long as we come from "cannot do" to "can do". Optimization can be pushed later instead of forcing the first patch to be perfectly completed.

> The deployment check requires more than 54G of free disk space, but after compiling, there is only about 12G of disk space left, which is not enough for correct deployment and operation.

Local development can still benefit from this config where I tested the compiled stuff takes 9.2G locally:

```
$ du -sh oceanbase | sort -rh 
9.6G	oceanbase
```

> requires more than 54G of free disk space

This can be optimized to be configured. I guess it's because OB reserved a large space for optimization. But for debugging purpose, we can use less space.

**Comment by caifeizhi** [comment]
> Cloud resources can be oversold. Locally it takes about 20 minutes to compile. I don't think it's a significant blocker as long as we come from "cannot do" to "can do". Optimization can be pushed later instead of forcing the first patch to be perfectly completed.
> 

Yes, as I mentioned earlier, the PR itself is fine and I will approve it. 

> This may be changed to be configurable. I guess it's because OB reserve a large space for optimization. But for debugging purposes, we can use less space.
some work has been done in https://github.com/oceanbase/oceanbase/issues/1363

**Comment by tisonkun** [comment]
@caifeizhi Thanks for your review and explanation!

**Review by caifeizhi — APPROVED** [review]

**Pull request merged**